### PR TITLE
updated examples to use new api

### PR DESCRIPTION
**Pull request by bbrock25**
_Why?_
Following the readme in the generator yields a broken example plugin.

_How?_
Updated the plugin example template to implement two methods
- echo: shows how to process input from the user and respond
- ping: demonstrates the hear method

_Side Effects?_
Shouldn't be any since this only touches template files.

### CODE REVIEW

**Comment by bbrock25** [comment]
also noticed that the help plugin isn't running by default, should we address that in this PR as well?

**Comment by keathley** [comment]
All new generated apps should include the help plugin. If you have an older version of the generator then it wouldn't be included by default.

File: generator/templates/new/plugins/example.ex [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
   use Butler.Plugin
 
-  respond(~r/example me/) do
-    reply "This is an example response"
+  @moduledoc """
+  replies to the user with an echo
+  """
+  respond ~r/echo (.*)/, conn, [_all, say] do
+
+    reply conn, "\necho: #{say}"
```
Comment (keathley) [inline generator/templates/new/plugins/example.ex]: You shouldn't need this newline here. Is the formatting broken somewhere?

**Comment by bbrock25** [comment]
Just verified that it was working with a fresh install via the console, I wonder what i did... Anyways, updated the pr slightly to demonstrate the code response as well.

File: generator/templates/new/plugins/example.ex [hunk 2]
```diff
@@ -1,11 +1,28 @@
...
+    # ["test"|tests]
+    say conn, "-> pong"
+  end
+
+  @moduledoc """
+  replies back with code; demonstrates escaping code blocks
+  """
+  respond ~r/code/, conn do
+    response = "console.log('this should look like code!');"
+      |> Responder.code
```
Comment (keathley) [inline generator/templates/new/plugins/example.ex:25]: You don't have to include the Responders. They're included with the Plugin module.

Comment (bbrock25) [inline generator/templates/new/plugins/example.ex:25]: so it would just be `|> code` ?

Comment (keathley) [inline generator/templates/new/plugins/example.ex:25]: Yup

**Comment by keathley** [comment]
I'm going to close this for now. If you still wanna update your branch and add some other examples plugins I'd love to see them.

**Pull request closed without merging**